rmc/utils/generic.py

```diff
@@ -289,50 +289,68 @@ def get_ref_aa(
     :param extra_aa_map: Dictionary mapping any amino acid one letter codes to three letter codes.
         Designed to capture any amino acids not present in `ACID_NAMES_PATH`.
         Default is {"*": "Ter", "U": "Sec"}.
-    :param aa_to_remove: Any amino acid one letter codes to remove. Default is "X".
-    :return: HT with one reference amino acid annotated per locus/transcript if `collect_all_aa` is False.
-        HT grouped by transcript with a list of all reference amino acids in that transcript if `collect_all_aa` is True.
+    :param aa_to_remove: Any amino acid one letter codes to remove. Default is {"X"}.
+    :return: HT with one reference amino acid annotated per locus/transcript.
     """
     logger.info("Mapping amino acid one letter codes to three letter codes...")
     aa_map = get_aa_map()
     # Add any extra mappings into amino acid mapping dict
     if extra_aa_map:
-        aa_map = aa_map.update(extra_aa_map)
-    ht = ht.annotate(ref_aa_1_letter=ht.amino_acids.split("/")[0])
-    ht = ht.annotate(
-        ref_aa=hl.literal(aa_map).get(ht.ref_aa_1_letter, ht.ref_aa_1_letter)
-    )
+        aa_map.update(extra_aa_map)
+    aa_map = hl.literal(aa_map)
+    ht = ht.annotate(ref_aa=aa_map.get(ht.ref_aa_1_letter, ht.ref_aa_1_letter))
```

Comment:
`ht` actually has the field `amino_acids`, not `ref_aa_1_letter`

Reply:
oh no, this must have gotten dropped when I applied a suggestion in vscode